identity.go

```diff
@@ -858,6 +861,10 @@ func ValidFor(id Identity, hostnameOrIp string) error {
 	if err != nil {
 		return &AddressError{BaseErr: ErrInvalidAddressForIdentity, Host: hostnameOrIp, ValidFor: getUniqueAddresses(id)}
 	}
```

Comment:
This block here is problematic too.

The loop will store the last return value in `err`

Meaning if we pass on the first one but fail on the second, we always fail.

The if statement checking err needs to be moved within the loop.

This function should probably have tests.